README.md

```diff
@@ -74,39 +44,174 @@ This program was created with the following tech stack and packages:
+| `avatar` |  String  | Specifies  avatar animation choice for video output. Options include default `barb` or non-default `boy` |
+
+#### API Constraints
+
+- This API only responds to POST calls
+- This process is computationally intensive, which may cause delay between requests. More user requests will result
+in slower processing times
+- The script text file should match the given audio file as closely as possible in terms of content. Failure to do so
+could lead to lip synchronization errors.
+- We recommend referencing Port 3000 in addition to the IP address.
+- File creation permissions must be allowed, or new files will not be created in `/tmp` directory.
+
+#### Possible API Errors
+
+| Error Code | Description |
+| :---      | :-----     |
+| 400 Bad Request | Required fields are specified |
+| 403 Forbidden | Request is recognized by the server but refused. Likely due to connecting to the wrong port |
+| 500 Internal Server Error | The API has not been configured properly on the host server machine |
+
+### The Phoneme Factory
+
+The Phoneme Factory is the first step in the SPP animation process. The factory is in charge of mapping spoken language
+from the `audio` and `text` inputs into a series of phonemes. Phonemes represent the smallest units of sound within
+words and help us distinguish when the mouth should change shape.
+
+To map these phonemes, the `audio` file, `text` file, and `recognizer` selection is passed to the **Phoneme processor**,
+which calls [**Rhubarb Lip Sync**](https://github.com/DanielSWolf/rhubarb-lip-sync). Rhubarb returns a JSON
+[output](#rhubarb-output) that outlines every phoneme along with a `start` and `end` time interval tag.
+
+#### Rhubarb Output
+
+```JSON
+"mouthCues": [
+    { "start": 0.00, "end": 0.08, "value": "X" },
+    { "start": 0.08, "end": 0.23, "value": "B" },
+    { "start": 0.23, "end": 0.37, "value": "C" },
+    { "start": 0.37, "end": 0.44, "value": "G" },
+    { "start": 0.44, "end": 1.42, "value": "B" },
+    ... ]
+
+```
+
+Rhubarb uses the `recognizer` input to select the engine to use to during the process of creating a phoneme file. When
+```recognizer: english```,  Rhubarb will use the `pocketSphinx` engine to process the english audio. When
+```recognizer: phonetic```, Rhubarb will use the `Phonetic` recognizer instead. It has been observed that the `Phonetic`
+recognizer performs better.
+
+Once the phoneme contents are created, they are sent to the next process.
+
+## The Animation Factory
+
+The animation factory uses the [phoneme output](#rhubarb-output) from the phoneme factory to logically select frames
+from `speaking-portal-project/images` to generate the 2D animation. Each frame corresponds to exactly one phoneme,
+avatar, blink value, and body position. This frame data is compiled into a [frame data output](#frame-data-output) file.
```

Comment:
I think these links are broken because when I tested it it kinda crashed my GitHub page but idk

Reply:
we also broke the "rhubarb-output" link, so I can double check and fix them

Reply:
fixed in another commit